routes/speech.rb

```diff
@@ -0,0 +1,26 @@
+
+class App < Sinatra::Base
+
+  get '/speeches' do
+    Speech.order(:id).limit(20).to_json
+  end
+
+  get '/getspeechbyuser/:user_id' do
+    Speech.where(user_id: params[:user_id]).to_json
+  end
+
+  get '/getspeaker/:speech_id' do
+    speech = Speech.find(params[:speech_id])
+    speech.speaker.to_json
+  end
+
+  get '/getaudiences/:speech_id' do
+    speech = Speech.find(params[:speech_id])
+    speech.audiences.to_json
+  end
+
+  get '/getparticipants/:speech_id' do
+    speech = Speech.find(params[:speech_id])
+    speech.participants.to_json
+  end
+end
```

Comment:
These URLs are bad naming.

Please learn more about "RESTful API", and how to design it.

http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api